New Tutorial RNA-seq STAR alignment (shell commands) (#4065)
galaxyproject/training-material: topics/transcriptomics/tutorials/rna-seq-bash-star-align/tutorial.md

>    - `GSM461177` (untreated): `GSM461177_1` and `GSM461177_2`
>    - `GSM461180` (treated): `GSM461180_1` and `GSM461180_2`
>
>    ```text
>    {{ page.zenodo_link }}/files/GSM461177_1.fastqsanger
>    {{ page.zenodo_link }}/files/GSM461177_2.fastqsanger
>    {{ page.zenodo_link }}/files/GSM461180_1.fastqsanger
>    {{ page.zenodo_link }}/files/GSM461180_2.fastqsanger
>    ```
>
>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
>    {% snippet faqs/galaxy/datasets_import_from_data_library.md %}
>
>    > <comment-title></comment-title>
>    >
>    > Note that these are the full files for the samples and ~1.5Gb each so it may take some minutes to import.
>    >
>    > For a quicker run-through of the FASTQ steps a small subset of each FASTQ file (~5Mb) can be found here on [Zenodo]({{ page.zenodo_link }}):
>    >
>    > ```text
>    > {{ page.zenodo_link }}/files/GSM461177_1_subsampled.fastqsanger
>    > {{ page.zenodo_link }}/files/GSM461177_2_subsampled.fastqsanger
>    > {{ page.zenodo_link }}/files/GSM461180_1_subsampled.fastqsanger
>    > {{ page.zenodo_link }}/files/GSM461180_2_subsampled.fastqsanger
>    > ```
>    >
>    {: .comment}
>

> >    >Change the datatype from `fastqsanger` to `fastq`.

Author: The right one should be fastq as it will make the mapping simpler.

> >    >Change the datatype from `fastqsanger` to `fastq`.

Reviewer: Yes sorry

>
>    {% snippet faqs/galaxy/datasets_change_datatype.md datatype="fastq.gz" %}
>
>
>    We also need to import two more files, essential for the alignment operation (and basically every alignment procedure): the organism's reference genome (here *D. melanogaster*) and the organism's gene annotation.

> >    Those can be aquired directly through data library as described above. For this tutorial we are going to use the files [dm3.fa](https://usegalaxy.eu/libraries/folders/Fdf740ff7e86f2efb/dataset/8ab324b9329be9fc) and [Drosophila_melanogaster.BDGP6.87.gtf (dm6)](https://usegalaxy.eu/libraries/folders/F30cab321d898d2fb/dataset/02c5f7fcdb6bf41f).

Reviewer: It does not make sense to use a fasta for dm3 and use a gtf for dm6.
You can get dm6 fasta at https://hgdownload.soe.ucsc.edu/goldenPath/dm6/bigZips/dm6.fa.gz

Author: You are absolutely right! However, I thing it would be better to "downgrade" the gtf file (to dm3) in order to have every piece of data available in Galaxy's library.

Reviewer: I think you can ask to have dm6.fa in a data library. For me it makes more sense.

Author: On second thought, it is instructed to fetch data from Zenodo a few lines above in the same tutorial, so my previous argument is invalid. Shall I apply the change you proposed in your previous comment, as quoted? 

> It does not make sense to use a fasta for dm3 and use a gtf for dm6.
> You can get dm6 fasta at https://hgdownload.soe.ucsc.edu/goldenPath/dm6/bigZips/dm6.fa.gz

>
> > <code-in-title>Environment and Packages</code-in-title>
> > ```bash
> > $ mkdir qualityRaw
> > $ fastqc /import/number_of_dataset_1_as_described_above -o qualityRaw/
> > $ fastqc /import/number_of_dataset_2_as_described_above -o qualityRaw/
> > $ fastqc /import/number_of_dataset_3_as_described_above -o qualityRaw/
> > $ fastqc /import/number_of_dataset_4_as_described_above -o qualityRaw/
> > ```
> {: .code-in}
>
{: .hands_on}

> <hands-on-title>Cutadapt</hands-on-title>
>
>  Now, it would be a good idea to trim the reads in order to get rid of bases that were sequenced with high uncertainty (i.e. low quality bases) at the read ends and also remove the reads of overall bad quality. We can also use this opportunity to restore the trimmed files names to make our job easier down the road. 
>
> > <code-in-title>Environment and Packages</code-in-title>
> > ```bash
> > $ mkdir trimmedData
> > $ cutadapt /import/2 /import/3_reverse_file_of_2 -o /trimmedData/GSM461177_R1 -p /trimmedData/GSM461177_R2
> > ```
> {: .code-in}

> > The same trimming procedure should take place for the second pair of reads (forwarld and reverse as above). After that, the files we are going to work with are the ones located in the **trimmedData** folder (4 in our case). 

Author: Thank you for catching those!

Image from ["Data Wrangling and Processing for Genomics"](https://datacarpentry.org/wrangling-genomics/)

We perform read alignment or mapping to determine where in the genome our reads originated from. There are a number of tools to choose from and, while there is no gold standard, there are some tools that are better suited for particular NGS analyses. We will be using the [STAR Aligner (STAR)](https://github.com/alexdobin/STAR), which is a universal aligner for mapping sequences against a large reference genome.

The alignment process consists of two steps:

1. Indexing the reference genome
2. Aligning the reads to the reference genome

# Setting up

> <hands-on-title>Index the reference genome</hands-on-title>
> Our first step is to index the reference genome for use by STAR. Indexing allows the aligner to quickly find potential alignment sites for query sequences in a genome, which saves time during alignment. Indexing the reference only has to be run once. The only reason you would want to create a new index is if you are working with a different reference genome or you are using a different tool for alignment.
>

> > > <code-in-title>Indexing with `bwa`</code-in-title>

Reviewer: ```suggestion
> > <code-in-title>Indexing with `STAR`</code-in-title>

```

> samples in our dataset (GSM461177). After that, we will be
> iterating the following process on the second pair of our sample files (GSM461180).
>
> > <code-in-title>Aligning with STAR</code-in-title>
> > ```bash
> > $ STAR --genomeDir ~/index --runThreadN 16 -- readFilesIn /trimmedData/GSM461177_R1 /trimmedData/GSM461177_R2 --outFileNamePrefix GSM461177
> > ```
> {: .code-in}
>
> > <code-out-title></code-out-title>
> > ```
> > Date Time ..... started STAR run
> > Date Time ..... loading genome
> > Date Time ..... started mapping
> > ```
> {: .code-out}
{: .hands_on}

# SAM/BAM format

The [SAM file](https://genome.sph.umich.edu/wiki/SAM),
is a tab-delimited text file that contains information for each individual read and its alignment to the genome. While we do not
have time to go into detail about the features of the SAM format, the paper by
[Heng Li et al.](http://bioinformatics.oxfordjournals.org/content/25/16/2078.full) provides a lot more detail on the specification.

**The compressed binary version of SAM is called a BAM file.** We use this version to reduce size and to allow for *indexing*, which enables efficient random access of the data contained within the file.

> The file begins with a **header**, which is optional. The header is used to describe the source of data, reference sequence, method of

Author: Thank you for pointing out. It should be clear now

---
layout: tutorial_hands_on
title: "RNA-seq Alignment with STAR"
questions:
  - "What are the first steps to process RNA-seq data?"
objectives:
  - "Learn the basics to process RNA sequences"
  - "Check the quality and trim the sequences with bash"
  - "Use command line STAR aligner to map the RNA sequences"
  - "Estimate the number of reads per gens"
time_estimation: "90M"
key_points:
  - "Bioinformatic command line tools are collections of commands that can be used to carry out bioinformatic analyses."
  - "To use most powerful bioinformatic tools, you will need to use the command line."
  - "There a few basic steps to follow in order to process the RNA sequences"
  - "Quality control, trimming and alignment to reference genome are the first part of the RNA-seq downstream analysis"

> contributors:

Reviewer 2: ```suggestion
contributions:
```
this should also fix the errors you are getting on this PR :)

- 3 treated samples (*Pasilla* gene depleted by RNAi): [GSM461179](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461179), [GSM461180](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461180), [GSM461181](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461181)

Each sample constitutes a separate biological replicate of the corresponding condition (treated or untreated). Moreover, two of the treated and two of the untreated samples are from a paired-end sequencing assay, while the remaining samples are from a single-end sequencing experiment.

> <comment-title>Full data</comment-title>
>
> The original data are available at NCBI Gene Expression Omnibus (GEO) under accession number [GSE18508](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE18508). The raw RNA-Seq reads have been extracted from the Sequence Read Archive (SRA) files and converted into FASTQ files.
{: .comment}

> <agenda-title></agenda-title>
>
> In this tutorial, we will cover:
>
> 1. TOC
> {:toc}
>
{: .agenda}

> <comment-title></comment-title>
>

> > This tutorial is significantly based on Galaxy's ["Reference-based RNA-Seq data analysis"](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html) tutorial.

Reviewer 3: ```suggestion
> This tutorial is significantly based on Galaxy's ["Reference-based RNA-Seq data analysis"]({% link topics/transcriptomics/tutorials/ref-based/tutorial.md %}) tutorial.

```

>
> > <code-out-title></code-out-title>
> > ```
> > STAR version: 2.7.10b   compiled: 2022-11-01T09:53:26-04:00 :/home/dobin/data/STAR/STARcode
> > STAR.master/source
> > Date Time ..... started STAR run
> > Date Time ... starting to generate Genome files
> > Date Time ..... processing annotations GTF
> > Date Time ... starting to sort Suffix Array. This may take a long time...
> > Date Time ... sorting Suffix Array chunks and saving them to disk...
> > ```
> {: .code-out}
>
{: .hands_on}

> <hands-on-title>Align reads to reference genome</hands-on-title>
>
> The alignment process consists of choosing an appropriate reference genome to map our reads against and then deciding on an aligner. We will use the STAR algorithm as said.
>
> An example of what a `STAR` command looks like is below. This command will not run, as we do not have the files: `input_file_R1` or `input_file_R2`.
>
>
> ```
> $ STAR --genomeDir /index_folder --runThreadN 16 --readFilesIn /trimmedData/input_file_R1 /trimmedData/input_file_R2
> ```
>
>
> Have a look at the [STAR manual](https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf), while we are running STAR with the default

> 

Reviewer 3: ```suggestion
>
```

this should fix the other issue

Author: Missed it! Thanks a lot for noticing!
